TrilinosWrappers::SolverDirect for Epetra data structures

peterrum opened the pull request:
Just in the case of `TrilinosWrappers::SolverBase` (and derived classes).

FYI @camillabelponer @jocgaray

bangerth reviewed (commented):
Would it be easy to create a test for this functionality?

jpthiele commented:
Should be quiite straightforward, e.g. by adapting the [Epetra tutorial](https://docs.trilinos.org/dev/packages/epetra/doc/html/Epetra_Lesson04.html) for a sparse matrix.

kronbichler reviewed (approved):
OK with me once we have a test.

peterrum commented:
Sorry for taking a so long. I have added a test. In addition, I have added a new function with the interface:
```cpp
    void
    solve(const SparseMatrix       &sparse_matrix,
          FullMatrix<double>       &solution,
          const FullMatrix<double> &rhs);
```
which directly calls the other new function.

marcfehling commented:
The test `trilinos/direct_solver_4.cc` fails in some configurations (see #18131).
